src/snowflake/snowpark/file_operation.py

```diff
@@ -219,8 +219,7 @@ def get(
                     self._session, plan
                 )._internal_collect_with_tag(statement_params=statement_params)
             return [GetResult(**file_result.asDict()) for file_result in get_result]
-        # connector raises IndexError when no file is downloaded from python connector.
-        except IndexError:
+        except OperationalError:
```

Comment:
The second test case in `test_get_negative_test` fails with OperationalError instead of IndexError. I looked into Python connector's code and couldn't find where it raises `IndexError` in file transfer related code.

Reply:
Also:  Looks like `test_get_negative_test` is skipped to avoid Azure/GCP-specific issues, we should still run it for AWS. Let's follow Python connector and use markers `@pytest.mark.aws` to select the corresponding tests in our tox config.

Reply:
This will be a behavior change

Reply:
`except OperationalError`? But I think this is the intended behavior: when no files are downloaded, just return a empty list and don't raise exceptions.

Reply:
yea, changing the exception type is usually a behavior change because some users might use try/except to catch the expected exception type.

Reply:
I suggest keeping the IndexError there. I remember I've seen IndexError before.

Reply:
Okay, I will revert this change for now, [SNOW-687929](https://snowflakecomputing.atlassian.net/browse/SNOW-687929) is created to track this fix. [SNOW-687933](https://snowflakecomputing.atlassian.net/browse/SNOW-687933) is created to track the proposal of using cloud provider specific pytest marks.